### ISSUE TITLE
Network in vsphere vm_attrs can only be configured by hidden/internal network title (aka. network_id)

### ISSUE THREAD

**Issue by cmeissner** [theforeman/foreman-ansible-modules#1041]
##### SUMMARY
For vsphere compute resources you can only set a network by it's internal network title (aka. network_id in attributes). This `network_id` is not as easy to guess as the `network_name`. 
The `network_id` will be generated internally by vsphere if the network is created. The `network_name` can be chosen by the administrators and so it can be predictable in contrast to `network_id`.

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION
```
2.1.2
```

##### COLLECTION VERSION
```
1.4.0
```

##### STEPS TO REPRODUCE
1. Create a compute profile
2. Configure vsphere compute resource within the newly created compute profile
3. run the following `cURL` command `curl -s -X GET --user <user>:<pass> https://foreman.example.com/api/compute_profiles/<id> | jq`
4. 
```json
{
  "id": "<id>",
  ...
  "vm_attrs": {
    ...
    "interfaces_attributes": {
      "0": {
        "type": "VirtualE1000",
        "network": "dvportgroup-31"
      },
    },
    ...
  },
  "attributes": {
    ...
    "interfaces_attributes": {
      "0": {
        "type_id": "VirtualE1000",
        "type_name": "E1000",
        "network_id": "dvportgroup-31",
        "network_name": "self_constructed_network_name"
      },
    }
  }
}
```
5. configure ansible task
```yaml
- name: "Manage compute profile"
  codeaffen.foreman.compute_profile:
    username: "{{ foreman_admin }}"
    password: "{{ foreman_admin_password }}"
    server_url: "https://foreman.example.com"
    validate_certs: "{{ validate_certs }}"
    name: "{{ compute_profile.name }}"
    compute_attributes:
      vm_attrs:
        ...
        interfaces_attributes:
          '0':
            type: VirtualE1000
            network: dvportgroup-30
        ...
    state: present
```

##### EXPECTED RESULTS
It would be great if `network_name` can be used over `network_id` in `vm_attrs`.

##### ACTUAL RESULTS
`network_id` has to be used as `network` in `vm_attrs`.

**Comment by mdellweg** [comment]
This looks like the interface attributes appear twice in the output of compute_profile. Is one of those a deprecated interface?

**Comment by evgeni** [comment]
I don't think so. The one (`vm_attrs`) is what we provide and the other (`attributes`) is what VMware does from it.

**Comment by evgeni** [comment]
Triage: look if Foreman could do the translation for us automatically, and if not implement a lookup using https://theforeman.org/api/2.1/apidoc/v2/compute_resources/available_networks.html

**Comment by evgeni** [comment]
Fixed in #1136
